packages/scene-composer/src/components/panels/scene-settings/OverlayPanelVisibilityToggle.tsx

```diff
@@ -49,14 +52,42 @@ export const OverlayPanelVisibilityToggle: React.FC = () => {
     [getSceneProperty, setSceneProperty, isViewing],
   );
 
+  const hasOverlayPanel = useMemo(() => {
+    const overlays = Object.keys(getComponentRefByType(KnownComponentType.DataOverlay));
+    return overlays.find(
+      (overlayRef) =>
+        (
+          findComponentByType(
+            getSceneNodeByRef(overlayRef),
+            KnownComponentType.DataOverlay,
+          ) as IDataOverlayComponentInternal
+        ).subType === Component.DataOverlaySubType.OverlayPanel,
+    );
+  }, [componentNodeMap, getComponentRefByType]);
```

Comment:
Where is componentNodeMap used?

Reply:
this is finding the overlay across all nodes, so any change to the map might mean an overlay added/removed